### PR TITLE
Update Go documentation

### PR DESCRIPTION
**Pull request by sealmove**
Before merging, please check if there are other outdated examples or documentation in general.

### CODE REVIEW

File: sdk/go/readme.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-(Tiny)Go. It currently allows building HTTP components that target the Wagi
+(Tiny)Go. It allows building HTTP components that target the Spin
 executor.
 
 ```go
 func main() {
- // call the HandleRequest function
- spin_http.HandleRequest(func(w http.ResponseWriter, r *http.Request) {
+ // call the Handle function
+ spin_http.Handle(func(w http.ResponseWriter, r *http.Request) {
```
Comment (radu-matei) [inline sdk/go/readme.md:10]: This example should be similar to https://github.com/fermyon/spin/blob/main/examples/http-tinygo/main.go.

Comment (sealmove) [reply sdk/go/readme.md:10]: Indeed, `main()` doesn't work now. `init()` is needed instead.

File: docs/content/go-components.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ allowed_http_hosts = [ "https://some-random-api.ml" ]
 route = "/hello"
 ```
 
-> Spin HTTP components written in Go must currently use the Wagi executor.
+> Spin HTTP components written in Go can now use the Spin executor.
```
Comment (itowlson) [inline docs/content/go-components.md:135]: nit: should probably drop the "now" (or even lose the para altogether?) - a reader not already familiar with the "old" way doesn't need to know that it used to be different

File: examples/http-tinygo-outbound-http/readme.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ source = "main.wasm"
 allowed_http_hosts = [ "https://some-random-api.ml", "https://postman-echo.com" ]
 [component.trigger]
 route = "/hello"
-executor = { type = "wagi" }
+executor = { type = "spin" }
```
Comment (itowlson) [inline examples/http-tinygo-outbound-http/readme.md:45]: Should we drop this since Spin is the default?

File: docs/content/go-components.md (hunk 2 above)
Comment (sealmove) [reply docs/content/go-components.md]: Fixed.

File: examples/http-tinygo-outbound-http/readme.md (hunk 3 above)
Comment (sealmove) [reply examples/http-tinygo-outbound-http/readme.md]: Dropped.

**Comment by radu-matei** [comment]
This looks great, thank you!

Could you please rebase (keep the useful commits), sign-off, and [add a commit GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)?

Thank you!

**Comment by sealmove** [comment]
> This looks great, thank you!
> 
> Could you please rebase (keep the useful commits), sign-off, and [add a commit GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)?
> 
> Thank you!

done

**Review by itowlson — APPROVED** [review]
Looks good - thanks!

(by the way, I fixed one tiny typo rather than going back to you again - hope that was okay)

**Comment by itowlson** [comment]
I'm really sorry about this but it looks like the GPG signatures are not working yet - our main branch is set up to need verified signatures which should show up like this:

image

https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification describes how to do this - if your commits are already GPG signed it may just be a matter of adding the key to your GitHub account, otherwise might need to run through the whole workflow.

Thanks for your patience!

**Comment by sealmove** [comment]
@itowlson I got my git setting straight! Thanks for the instructions. I squashed everything in one verified commit.

**Review by radu-matei — APPROVED** [review]

**Pull request merged**